cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/startup/WorkerApplication.java

```diff
@@ -60,65 +60,65 @@
 import se.simonsoft.cms.version.CmsComponents;
 
 public class WorkerApplication extends ResourceConfig {
-	
+
 	private final Environment environment = new Environment();
-	
+
 	private static final String AWS_ACTIVITY_NAME = "abxpe";
 	private static final String PUBLISH_FS_PATH_ENV = "PUBLISH_FS_PATH";
 	private static final String PUBLISH_S3_BUCKET_ENV = "PUBLISH_S3_BUCKET";
 	private static final String BUCKET_NAME = "cms-automation";
-	
+
 	private final CmsExportPrefix exportPrefix = new CmsExportPrefix("cms4");
-	
+
 	private Region region; // The Region class will be in SDK 2.0 (Regions will be removed).
-	private String cloudId; 
+	private String cloudId;
 	private String awsAccountId;
 	private AWSCredentialsProvider credentials = DefaultAWSCredentialsProviderChain.getInstance();
-	private String bucketName = BUCKET_NAME; 
+	private String bucketName = BUCKET_NAME;
 	private ServletContext context;
-	
+
 	private static final Logger logger = LoggerFactory.getLogger(WorkerApplication.class);
 
 	private static CmsComponentVersion webappVersion;
 
 	public WorkerApplication(@Context ServletContext context)  {
-		
+
 		this.context = context;
 		logger.info("Worker Webapp starting with context: " + context);
-		
+
 		setWebappVersion(context);
 		context.setAttribute("buildName", getWebappVersionString());
 		CmsComponents.logAllVersions();
-		
+
 		register(new AbstractBinder() {
 
 			@Override
             protected void configure() {
-            	
+
 				String aptapplicationPrefix = getAptapplicationPrefix();
-				
-            	bind(new PublishServicePe()).to(PublishServicePe.class);
-            	
-            	PublishServicePe publishServicePe = new PublishServicePe();
-            	PublishJobService publishJobService = new PublishJobService(publishServicePe, aptapplicationPrefix);
-            	bind(publishJobService).to(PublishJobService.class);
-            	
+
             	WorkerStatusReport workerStatusReport = new WorkerStatusReport();
             	bind(workerStatusReport).to(WorkerStatusReport.class);
-            	
+
             	String envBucket = environment.getParamOptional(PUBLISH_S3_BUCKET_ENV);
             	if (envBucket != null) {
             		logger.debug("Will use bucket: {} specified in environment", envBucket);
             		bucketName = envBucket;
             	}
             	bind(bucketName).named("config:se.simonsoft.cms.publish.bucket").to(String.class);
             	cloudId = environment.getParamOptional("CLOUDID");
-            	
-            	region = Region.getRegion(Regions.fromName("eu-west-1")); // Currently hardcoded, might need different regions annotated per service. 
-        		// Might need to determine which region we are running in for EC2. See SDK 2.0, might have a Region Provider Chain.
-        		bind(region).to(Region.class);
+
+            	region = Region.getRegion(Regions.fromName("eu-west-1")); // Currently hardcoded, might need different regions annotated per service.
```

Comment:
Oh, this is an old sin that I might be the author of. Can you try using DefaultAwsRegionProviderChain instead?

Reply:
No problem!